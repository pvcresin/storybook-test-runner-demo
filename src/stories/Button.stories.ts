import type { Meta, StoryObj } from "@storybook/react";
import { fn, expect, userEvent, within } from "@storybook/test";

import { Button } from "./Button";

const meta = {
  title: "Example/Button",
  component: Button,
  parameters: {
    layout: "fullscreen", // https://storybook.js.org/docs/configure/story-layout
  },
  tags: ["autodocs"], // https://storybook.js.org/docs/writing-docs/autodocs
  argTypes: {
    backgroundColor: { control: "color" }, // https://storybook.js.org/docs/api/argtypes
  },
  args: {
    onClick: fn(), // https://storybook.js.org/docs/essentials/actions#action-args
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    primary: true,
    label: "Button",
  },
};

export const Play: Story = {
  args: {
    primary: false,
    label: "Button",
  },
  // https://storybook.js.org/docs/writing-tests/interaction-testing
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const loginButton = canvas.getByRole("button", { name: /Button/i });
    await expect(loginButton).toBeInTheDocument();
    await userEvent.click(loginButton);
  },
};
