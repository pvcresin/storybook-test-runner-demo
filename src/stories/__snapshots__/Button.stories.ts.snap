// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/Button Play play-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--medium storybook-button--secondary"
>
  Button
</button>
`;

exports[`Example/Button Primary smoke-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--medium storybook-button--primary"
>
  Button
</button>
`;
